scripts/atlas/lib/emit.mjs

```
@@ -31,20 +37,63 @@ export function normalizedUtf8ByteLength(data) {
+      maxBytes: Math.max(Buffer.byteLength(next, 'utf8') + 1, 16 * 1024 * 1024),
+    });
+    if (written !== next) {
+      throw new Error(`Atlas output verification failed: ${relativePath}`);
+    }
+    return true;
+  } finally {
+    if (descriptor !== undefined) fs.closeSync(descriptor);
+    try {
+      fs.unlinkSync(temporaryPath);
+    } catch (error) {
+      if (error?.code !== 'ENOENT') throw error;
```

Biome (2.5.7)

[error] Unsafe usage of 'throw'.

(lint/correctness/noUnsafeFinally)